target_bigquery/core.py

```diff
@@ -500,9 +500,9 @@ def clean_up(self) -> None:
             if self._is_dedupe_before_upsert_candidate():
                 # We can't use MERGE with a non-unique key, so we need to dedupe the temp table into
                 # a _SESSION scoped intermediate table.
-                tmp = f"{target.name}__tmp"
+                tmp = f"{self.merge_target.name}__tmp"
```

Comment:
`target` is of `Table` type that doesn't have a `name` property, `merge_target` (`BigQueryTable` type) has.